code/game/objects/structures/traps.dm

```diff
@@ -81,10 +86,84 @@
 	name = "shock trap"
 	desc = "A trap that will shock and render you immobile. You'd better avoid it."
 	icon_state = "trap-shock"
+	var/stun_time = 100
 
 /obj/structure/trap/stun/trap_effect(mob/living/L)
 	L.electrocute_act(30, src, safety=1) // electrocute act does a message.
-	L.Paralyze(100)
+	L.Paralyze(stun_time)
+
+/obj/structure/trap/stun/hunter
+	name = "bounty trap"
+	desc = "A trap that only goes off when a fugitive steps on it, announcing the location and stunning the target. You'd better avoid it."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "bounty_trap_on"
+	stun_time = 200
+	sparks = FALSE //the item version gives them off to prevent runtimes (see Destroy())
+	checks_antimagic  = FALSE
+	var/obj/item/bountytrap/stored_item
+	var/caught = FALSE
+
+/obj/structure/trap/stun/hunter/Initialize(mapload)
+	. = ..()
+	time_between_triggers = 10
+	flare_message = "<span class='warning'>[src] snaps shut!</span>"
+
+/obj/structure/trap/stun/hunter/Crossed(atom/movable/AM)
+	if(isliving(AM))
+		var/mob/living/L = AM
+		if(!L.mind?.has_antag_datum(/datum/antagonist/fugitive))
+			return
+	caught = TRUE
+	. = ..()
+
+/obj/structure/trap/stun/hunter/flare()
+	..()
+	stored_item.forceMove(get_turf(src))
+	forceMove(stored_item)
+	if(caught)
+		stored_item.announce_fugitive()
+		caught = FALSE
+
+/obj/item/bountytrap
+	name = "bounty trap"
+	desc = "A trap that only goes off when a fugitive steps on it, announcing the location and stunning the target. It's currently inactive."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "bounty_trap_off"
+	var/obj/structure/trap/stun/hunter/stored_trap
+	var/obj/item/radio/radio
+	var/datum/effect_system/spark_spread/spark_system
+
+/obj/item/bountytrap/Initialize(mapload)
+	. = ..()
+	radio = new(src)
+	radio.subspace_transmission = TRUE
+	radio.canhear_range = 0
+	radio.recalculateChannels()
+	spark_system = new
+	spark_system.set_up(4,1,src)
+	spark_system.attach(src)
+	name = "[name] #[rand(1, 999)]"
+	stored_trap = new(src)
+	stored_trap.name = name
+	stored_trap.stored_item = src
+
+/obj/item/bountytrap/proc/announce_fugitive()
+	spark_system.start()
+	playsound(src, 'sound/machines/ding.ogg', 50, 1)
+	radio.talk_into(src, "Fugitive has triggered this trap in the [get_area_name(src)]!", RADIO_CHANNEL_COMMON)
+
+/obj/item/bountytrap/attack_self(mob/living/user)
+	if(!user || !user.temporarilyRemoveItemFromInventory(src))//visibly unequips
```

Comment:
why not just https://github.com/tgstation/tgstation/blob/412c3b63e39c60ecd1dfdfa245e040e92b4ff59b/code/modules/mob/inventory.dm#L300

Reply:
sure